Cosminis/DataAccess/FriendsRepo.cs

```diff
@@ -0,0 +1,457 @@
+
+        if(friendsQuery == null)
+        {
+            throw new ResourceNotFound("This is one lonely user. At this point they should simply quit social media.");
+        }
+
+        try
+        {       
+            List<Friends> friendsList = friendsQuery.ToList();
+            
+            if(friendsList.Count() < 1)
+            {
+                throw new ResourceNotFound();
+            }
+
+            return friendsList;
+        }
+        catch(Exception E)
+        {
+            throw;
+        }                                  
+    }
+    
+    public Friends FriendsByUserIds(int searchingUserId, int user2BeSearchedFor)
+    {
+        User searchingUser = _context.Users.Find(searchingUserId);
+        User friend2BFound = _context.Users.Find(user2BeSearchedFor);
+
+        List<Friends> quieriedFriendsList = ViewAllFriends((int)searchingUser.UserId);
+
+        List<Friends> searchedFriendsList = ViewAllFriends((int)friend2BFound.UserId);
+
+        Friends returnRelationship = new Friends();
+
+        foreach(Friends friendInstance in quieriedFriendsList)
+        {
+            if(searchedFriendsList.Contains(friendInstance))
+            {
+                returnRelationship = friendInstance;
+            }
+        }
+        return returnRelationship;
+    }
+    
+    public Friends EditFriendship(int editingUserID, int user2BeEdited, string status)//this does not delete removed/blocked friends. I may add this later.
+    {   
+        User editingUser = _context.Users.Find(editingUserID);
+        User friend2BeEdited = _context.Users.Find(user2BeEdited);
+        
+        Friends statusToBeEdited = FriendsByUserIds((int)editingUser.UserId, (int)friend2BeEdited.UserId);
+
+        statusToBeEdited.Status = status;
+
+        _context.SaveChanges();
+
+        _context.ChangeTracker.Clear(); 
+
+        return statusToBeEdited;  
+    }
+
+    public Friends SearchByRelationshipId(int relationshipId)
+    {
+        return _context.Friends.FirstOrDefault(friends => friends.RelationshipId == relationshipId) ?? throw new ResourceNotFound();
+    }
+
+    public List<Friends> CheckRelationshipStatusByUsername(string username, string status)
+    {   
+        User quieriedUser = _userRepo.GetUserByUserName(username);
+
+        List<Friends> quieriedUsersList = ViewAllFriends((int)quieriedUser.UserId);
+
+        List<Friends> statusList = new List<Friends>();
+
+        if(quieriedUsersList == null)
+        {
+            throw new ResourceNotFound();
+        }
+        foreach(Friends friendSearch in quieriedUsersList)
+        {
+            if(friendSearch.Status.ToString() == status)
+            {
+                statusList.Add(friendSearch);
+            }
+        }
+        if(statusList.Count() < 1)
+        {
+            throw new ResourceNotFound();
+        }
+
+        return statusList;
+    }
+
+    public List<Friends> CheckRelationshipStatusByUserId(int searchingId, string status)
```

Comment:
Please document your methods, I am not going to read anything forwards because it is straining my brain to go over your code line by line

Reply:
I'll add in comments as we proceed with the project and try to retroactively comment my code when I have time.